pkg/restore/restore_test.go

```diff
@@ -2113,6 +2112,130 @@ func TestRestorePersistentVolumes(t *testing.T) {
 	}
 }
 
+type fakeResticRestorerFactory struct {
+	restorer         *resticmocks.Restorer
+	podVolumeBackups []*velerov1api.PodVolumeBackup
+}
+
+func (f *fakeResticRestorerFactory) NewRestorer(ctx.Context, *velerov1api.Restore) (restic.Restorer, error) {
+	return f.restorer, nil
+}
+
+type fakeResticRestorer struct {
+	podVolumeBackups []*velerov1api.PodVolumeBackup
+}
+
+func (b *fakeResticRestorer) RestorePodVolumes(data restic.RestoreData) []error {
+	return nil
+}
+
+// TestRestoreWithRestic runs restores of only pods that contain associated PVBs.
+func TestRestoreWithRestic(t *testing.T) {
+	tests := []struct {
+		name                        string
+		restore                     *velerov1api.Restore
+		backup                      *velerov1api.Backup
+		apiResources                []*test.APIResource
+		podVolumeBackups            []*velerov1api.PodVolumeBackup
+		podWithPVBs, podWithoutPVBs []*corev1api.Pod
+	}{
+		{
+			name:         "a pod that exists in given backup and contains associated PVBs should have restored pods",
```

Comment:
maybe instead of "should have restored pods", something like "should have RestorePodVolumes called"?